js/CameraRoll.js

```diff
@@ -186,27 +205,37 @@ class CameraRoll {
   ): Promise<Album[]> {
     return RNCCameraRoll.getAlbums(params);
   }
+
+  static getParamsWithDefaults(params: GetPhotosParams): GetPhotosParams {
+    const newParams = {...params};
+    if (!newParams.assetType) {
+      newParams.assetType = ASSET_TYPE_OPTIONS.All;
+    }
+    if (!newParams.groupTypes && Platform.OS !== 'android') {
+      newParams.groupTypes = GROUP_TYPES_OPTIONS.All;
+    }
+    return newParams;
+  }
+
   /**
    * Returns a Promise with photo identifier objects from the local camera
    * roll of the device matching shape defined by `getPhotosReturnChecker`.
    *
    * See https://facebook.github.io/react-native/docs/cameraroll.html#getphotos
    */
   static getPhotos(params: GetPhotosParams): Promise<PhotoIdentifiersPage> {
-    if (!params.assetType) {
-      params.assetType = ASSET_TYPE_OPTIONS.All;
-    }
-    if (!params.groupTypes && Platform.OS !== 'android') {
-      params.groupTypes = GROUP_TYPES_OPTIONS.All;
-    }
     if (arguments.length > 1) {
       console.warn(
         'CameraRoll.getPhotos(tag, success, error) is deprecated.  Use the returned Promise instead',
       );
       let successCallback = arguments[1];
       const errorCallback = arguments[2] || (() => {});
-      RNCCameraRoll.getPhotos(params).then(successCallback, errorCallback);
```

Comment:
This is actually correct I think. In the previous version of the code, it makes this call. And then, since there isn't an early return, it then makes a duplicate call and returns the promise anyways. This code maintains the same behaviour and return value as before, but only calls the native getPhotos once.

Reply:
Ah you are right that it falls through to the second call. I guess it doesn't hurt to return the promise. Alternatively you could return without a value, but I'll leave that up to you.